mkosi/__init__.py

```diff
@@ -7366,7 +7366,7 @@ def suppress_stacktrace() -> Iterator[None]:
         yield
     except subprocess.CalledProcessError as e:
         # MkosiException is silenced in main() so it doesn't print a stacktrace.
-        raise MkosiException(e)
+        raise MkosiException(e, returncode=e.returncode)
```

Comment:
If we chain the exceptions like this
```suggestion
        raise MkosiException(e) from e
```
we can do something like this in main:
```python
except MkosiException as e:
    cause = getattr(e, "__cause__", None)
    if cause:
        if isinstance(cause, CalledProcessError):
            if args.verb == "build":
                print(cause.args)
            sys.exit(cause.returncode)
    sys.exit(1)
```

This way we would reuse the contextual information we already have from the initial exception and MkosiException could be a thin wrapper around wherever we might use it.

Reply:
I'm still not 100% convinced. We tend to print errors when they happen and before all the finalizing messages get printed. This would change that behavior. I figure if we want to start printing cmdlines, we should do so right before we raise the exception and print all the finalizer messages instead of after all the finalizer messages. For now, let's merge this as is since this is a slightly different issue than the one we're trying to fix here.